src/os/filestore/LFNIndex.cc

```diff
@@ -429,7 +429,18 @@ int LFNIndex::list_objects(const vector<string> &to_list, int max_objs,
   int r = 0;
   int listed = 0;
   bool end = true;
-  while ((de = ::readdir(dir))) {
+  while (true) {
+    errno = 0;
+    de = ::readdir(dir);
+    if (de == nullptr) {
+      if (errno != 0) {
+        r = -errno;
+        dout(0) << "readdir failed " << to_list_path.c_str() << ": "
```

Comment:
nit: no need to use `c_str`. Just `<< to_list_path` would be fine.

Reply:
fixed this too. thanks.